packages/backend/src/open_payments/payment/incoming/service.test.ts

```diff
@@ -473,6 +473,62 @@ describe('Incoming Payment Service', (): void => {
       ).resolves.toBe(IncomingPaymentError.InvalidExpiry)
     })
   })
+  describe('Update IncomingPayment', (): void => {
+    let amount: Amount
+    let payment: IncomingPayment
+
+    beforeEach(async (): Promise<void> => {
+      amount = {
+        value: BigInt(123),
+        assetCode: asset.code,
+        assetScale: asset.scale
+      }
+      payment = (await incomingPaymentService.create({
+        walletAddressId,
+        incomingAmount: amount
+      })) as IncomingPayment
+      assert.ok(!isIncomingPaymentError(payment))
+    })
+
+    test.each`
+      metadata
+      ${{ description: 'Update metadata', status: 'COMPLETE' }}
+      ${{}}
+    `('An incoming payment can be updated', async (metadata): Promise<void> => {
+      await expect(
+        IncomingPaymentEvent.query(knex).where({
+          type: IncomingPaymentEventType.IncomingPaymentUpdated
+        })
+      ).resolves.toHaveLength(0)
+
+      const incomingPayment = await incomingPaymentService.update({
+        id: payment.id,
+        metadata
+      })
+      assert.ok(!isIncomingPaymentError(incomingPayment))
+      expect(incomingPayment).toMatchObject({
+        id: incomingPayment.id,
+        metadata
+      })
+      await expect(
+        IncomingPaymentEvent.query(knex).where({
+          type: IncomingPaymentEventType.IncomingPaymentUpdated
+        })
+      ).resolves.toHaveLength(0)
+    })
```

Comment:
```suggestion
    `('An incoming payment can be updated', async ({ metadata }): Promise<void> => {
      const incomingPayment = await incomingPaymentService.update({
        id: payment.id,
        metadata
      })
      assert.ok(!isIncomingPaymentError(incomingPayment))
      expect(incomingPayment).toMatchObject({
        id: incomingPayment.id,
        metadata
      })
    })
```

Reply:
we don't need to worry about `IncomingPaymentEvent`, we are not creating an event in the `.update` function, just updating the Incoming Payment model